nexus/tests/integration_tests/volume_management.rs

```diff
@@ -2552,3 +2559,619 @@ async fn test_volume_hard_delete_idempotent(
     datastore.volume_hard_delete(volume_id).await.unwrap();
     datastore.volume_hard_delete(volume_id).await.unwrap();
 }
+
+// upstairs related tests
+
+/// Test that an Upstairs can reissue live repair notifications
+#[nexus_test]
+async fn test_upstairs_repair_notify_idempotent(
+    cptestctx: &ControlPlaneTestContext,
+) {
+    let int_client = &cptestctx.internal_client;
+
+    let upstairs_id: TypedUuid<UpstairsKind> = TypedUuid::new_v4();
+    let session_id: TypedUuid<UpstairsSessionKind> = TypedUuid::new_v4();
+    let repair_id: TypedUuid<UpstairsRepairKind> = TypedUuid::new_v4();
+    let region_id: TypedUuid<DownstairsRegionKind> = TypedUuid::new_v4();
+
+    // Notify start
+    let notify_url = format!("/upstairs/{upstairs_id}/repair-start");
+
+    int_client
+        .make_request(
+            Method::POST,
+            &notify_url,
+            Some(internal::nexus::RepairStartInfo {
```

Comment:
Nit: it seems like this `internal::nexus::RepairStartInfo` and the one called
below are the same.  Could we pull this out to a variable and call the
`make_request` with it like that?  It would be easier to see right away that
the test was sending the same content twice (and fewer lines of code :) )

Reply:
done in 3dfabf027